consumer.go

```diff
@@ -145,19 +174,37 @@ func (c *consumer) LastModified() time.Time {
 }
 
 // Close implements Consumer interface.
-func (c *consumer) Close() error {
+func (c *consumer) Close() (err error) {
 	c.stop()
 	if c.ownRemote {
-		return c.remote.Close()
+		if e := c.remote.Close(); e != nil {
+			err = e
+		}
+		if c.bucket != nil {
```

Comment:
this should not be within the `ownRemote` block but be: `if c.ownBucket && c.bucket != nil`